frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1809,40 +1809,46 @@ function ReaderFooter:onExitFlippingMode()
 end
 
 function ReaderFooter:onTapFooter(ges)
-    if self.has_no_mode or self.settings.lock_tap then
-        return
+    local ignore_lock = false
+    if ges == true then
+        ignore_lock = true
+        ges = nil
     end
-    if self.view.flipping_visible then
+    if self.view.flipping_visible and ges then
         local pos = ges.pos
         local dimen = self.progress_bar.dimen
         -- if reader footer is not drawn before the dimen value should be nil
         if dimen then
             local percentage = (pos.x - dimen.x)/dimen.w
             self.ui:handleEvent(Event:new("GotoPercentage", percentage))
         end
-    else
-        if self.settings.all_at_once or self.has_no_mode then
-            if self.mode >= 1 then
-                self.mode = self.mode_list.off
-            else
-                self.mode = self.mode_list.page_progress
-            end
+        self:updateFooter(true)
+        return true
+    end
+    if self.has_no_mode or (self.settings.lock_tap and not ignore_lock) then
+        return
+    end
+    if self.settings.all_at_once or self.has_no_mode then
+        if self.mode >= 1 then
+            self.mode = self.mode_list.off
         else
-            self.mode = (self.mode + 1) % self.mode_nb
-            for i, m in ipairs(self.mode_index) do
-                if self.mode == self.mode_list.off then break end
-                if self.mode == i then
-                    if self.settings[m] then
-                        break
-                    else
-                        self.mode = (self.mode + 1) % self.mode_nb
-                    end
+            self.mode = self.mode_list.page_progress
+        end
+    else
+        self.mode = (self.mode + 1) % self.mode_nb
```

Comment:
That's the remainder of available modes?

Reply:
(Original code not changed, it just appears in the diff because of my change in if/else indentation.)
I guess it's just a way to cycle thru all modes, and when reaching end of list, go back to start - which is not `1` because we have a mode 0 = off :)
(It' s a bit of magic mixed with mess :) See https://github.com/koreader/koreader/issues/6210#issuecomment-636150143. )